packages/storage/src/components/UpdateUserPhotoModal/index.js

```diff
@@ -0,0 +1,233 @@
+
+    const photoURL = user && user.photoURL ? user.photoURL : null;
+    const initials = user && user.displayName ? getInitials(user.displayName) : null;
+
+    const pathProp = path || `users/${user ? user.uid : ''}/profilePhoto`;
+    const bucketProp = bucket || `${app.options.storageBucket}`;
+
+    const cropperRef = useRef(null);
+    const [uploadedImage, setUploadedImage] = useState();
+    const [isOpenEdit, setIsOpenEdit] = useState(false);
+    const [isLoading, setIsLoading] = useState(false);
+    const [error, setError] = useState();
+
+    const handleOnClickEdit = () => {
+        setIsOpenEdit(true);
+    };
+
+    const onCrop = async () => {
+        const imageElement = cropperRef ? cropperRef.current : null;
+        const cropper = imageElement ? imageElement.cropper : null;
+        const storage = getStorage(app, bucketProp);
+        const photoRef = ref(storage, `${pathProp}/${nanoid()}`);
+        const blob = await (await fetch(cropper.getCroppedCanvas().toDataURL())).blob();
+
+        const uploadTask = uploadBytesResumable(photoRef, blob);
+        uploadTask.on(
+            'state_changed',
+            (snapshot) => {
+                const progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
+                if (progress !== 100) {
+                    setIsLoading(true);
+                }
+            },
+            (error) => {
+                setIsLoading(false);
+                setError(error.message);
+            },
+            async () => {
+                const photoURL = await getDownloadURL(photoRef);
+                await updateProfile(user, {
+                    photoURL,
+                });
+                await reload();
+                setError(null);
+                setIsOpenEdit(false);
+                setIsLoading(false);
+                onPhotoUpdated(photoURL);
+                onRequestClose();
+            },
+        );
+    };
+
+    const onClickRotate = () => {
+        const imageElement = cropperRef ? cropperRef.current : null;
+        const cropper = imageElement ? imageElement.cropper : null;
+        cropper.rotate(90);
+    };
+
+    const handleOnClickRemove = async () => {
+        const result = await confirmModal({
+            icon: <TrashIcon />,
+            variant: 'destructive',
+            header: 'Delete Content',
+            question: "This item will be deleted immediately. You can't undo this action.",
+            okButtonLabel: 'Delete',
+        });
+        if (result) {
+            const storage = getStorage(app);
+            const photoRef = ref(storage, photoURL);
+            try {
+                await deleteObject(photoRef);
+                await updateProfile(user, {
+                    photoURL: '',
+                });
+                setIsOpenEdit(false);
+                await reload();
+                onPhotoUpdated(null);
+            } catch (error) {
+                showAppMessage({
```

Comment:
look at the design that tahimi attached, it is not an app message but a custom error message inside the modal